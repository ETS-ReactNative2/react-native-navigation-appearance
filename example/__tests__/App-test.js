import 'react-native'
import React from 'react'
import { setDarkModeMock } from '@busfor/react-native-navigation-appearance'
import AppScreen from '../src/AppScreen'

import renderer from 'react-test-renderer'

it('light theme', () => {
  setDarkModeMock(false)
  const tree = renderer.create(<AppScreen />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('dark theme', () => {
  setDarkModeMock(true)
  const tree = renderer.create(<AppScreen />).toJSON()
  expect(tree).toMatchSnapshot()
})
