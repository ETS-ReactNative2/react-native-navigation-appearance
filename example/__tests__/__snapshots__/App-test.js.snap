// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dark theme 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#121212",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#fff",
          "paddingVertical": 8,
          "textAlign": "center",
        }
      }
    >
      appearance: 
      dark
    </Text>
    <Text
      style={
        Object {
          "color": "#fff",
          "paddingVertical": 8,
          "textAlign": "center",
        }
      }
    >
      current mode: 
      light
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#fff",
        "paddingVertical": 8,
        "textAlign": "center",
      }
    }
  >
    Choose mode:
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 24,
      }
    }
  >
    <View
      style={
        Object {
          "marginHorizontal": 8,
        }
      }
    >
      <View
        style={
          Object {
            "marginHorizontal": 8,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              system
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            light
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginHorizontal": 8,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            dark
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`light theme 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#121212",
          "paddingVertical": 8,
          "textAlign": "center",
        }
      }
    >
      appearance: 
      light
    </Text>
    <Text
      style={
        Object {
          "color": "#121212",
          "paddingVertical": 8,
          "textAlign": "center",
        }
      }
    >
      current mode: 
      light
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#121212",
        "paddingVertical": 8,
        "textAlign": "center",
      }
    }
  >
    Choose mode:
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 24,
      }
    }
  >
    <View
      style={
        Object {
          "marginHorizontal": 8,
        }
      }
    >
      <View
        style={
          Object {
            "marginHorizontal": 8,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              system
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            light
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginHorizontal": 8,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            dark
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
